pandas/core/indexes/period.py

```diff
@@ -517,6 +518,10 @@ def astype(self, dtype, copy=True, how='start'):
             return self.to_timestamp(how=how).tz_localize(dtype.tz)
         elif is_period_dtype(dtype):
             return self.asfreq(freq=dtype.freq)
+        elif is_categorical_dtype(dtype):
```

Comment:
same

side thing, I *think* that we could make a more generic astype in  ``indexes.base`` and remove *some* boiler plate *maybe* (of course separate PR), you can make an issue if you want (or just PR!)